app/controllers/ui_ansible_roles_controller.rb

```diff
@@ -7,6 +10,12 @@ def index
     @ui_ansible_roles = resource_scope_for_index(:permission => :view_ansible_roles)
   end
 
+  api :DELETE, '/ui_ansible_roles/:id', N_('Deletes Ansible role')
```

Comment:
I don't see any permission filtering here. Am I missing something?

Also, couldn't this just use the regular REST API to delete roles?

Reply:
Indeed, I forgot to add the permission.

The table component uses the same endpoint for querying and deleting items, so it's unfortunately not possible to use the API just for deletion.

Reply:
Unfortunately, this would require significant changes to TableIndexPage.js and Table.js in core, as the idea is that TableIndexPage represents an interface to a single controller that has the necessary CRUD methods.
One could add another prop, `deletionURL` to TableIndexPage and Table that would take precedence over the original URL for deletion but I am not sure that is a cleaner solution.

Reply:
Can't we add the `Delete` option to the `rowKebabItems` action items instead of using the `isDeleteable` property? From there, we can use the URL of the existing API for the delete action.

Reply:
Unfortunately, we would lose the delete modal that way.